Update Helm 3 doc to improve usability (#27598)
istio/istio: manifests/charts/README-helm3.md

# Helm v3 support

## Install

> The Helm charts are support both by Helm v2 and Helm v3. Please do not introduce Helm v3 specific changes as many

Reviewer: ```suggestion
The Helm charts are supported both by Helm v2 and Helm v3. Please do not introduce Helm v3 specific changes as many
```

Author: Updated

users are still using Helm v2 and the operator is currently using the Helm v2 code to generate.

To install with Helm v3, you must first create the namespace that you wish to install in if the namespace does not exist already. The default namespace used is `istio-system` and can be created as follows:

```console
kubectl create namespace istio-system
```

The charts are as follows:

> - 'base' creates cluster-wide CRDs, cluster bindings and cluster resources. It is possible to change the namespace from `istio-system` but it is not recommended.

Reviewer: ```suggestion
- `base` creates cluster-wide CRDs, cluster bindings, and cluster resources. It is possible to change the namespace from the `istio-system` but it is not recommended.
```

Author: Done

```console
helm install istio-base -n istio-system manifests/charts/base
```

> - 'istio-control/istio-discovery' installs a revision of istiod.  You can install it multiple times, with different revisions.

Reviewer: ```suggestion
- `istio-control/istio-discovery` installs a revision of `istiod`.  You can install it multiple times, with different revisions.
```

Author: Done

```console

>  helm install -n istio-system istio-17 manifests/charts/istio-control/istio-discovery -f manifests/charts/global.yaml

Reviewer 2: I thought we removed the requirement for this. @jacob-delgado is it still required, and if not can it be removed?

Reviewer 3: It should be removed after https://github.com/istio/istio/pull/25186

Author: That is a typo by me, apologies. It is working without the global values.

Author: Removed.

 helm install -n istio-system istio-canary manifests/charts/istio-control/istio-discovery \
    -f manifests/charts/global.yaml  --set revision=canary --set clusterResources=false

 helm install -n istio-system istio-mytest manifests/charts/istio-control/istio-discovery \
    -f manifests/charts/global.yaml  --set revision=mytest --set clusterResources=false
```

> - 'gateways' install a load balancer with `ingress` and `egress`. You can install it multiple times with different revisions but they must be installed in separate namespaces.

Reviewer: ```suggestion
- `gateways` install a load balancer with `ingress` and `egress`. You can install it multiple times with different revisions but they must be installed in separate namespaces.
```

Author: Done
